src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1367,7 +1386,13 @@ static bool bindMethodGroupInvocation(
 
                 if (!methodGroup.HasAnyErrors) diagnostics.AddRange(resolution.Diagnostics); // Suppress cascading.
 
-                if (resolution.HasAnyErrors)
+                if (resolution.IsNonMethodExtensionMember(out Symbol? extensionMember))
+                {
+                    ReportMakeInvocationExpressionBadMemberKind(syntax, WellKnownMemberNames.CollectionInitializerAddMethodName, methodGroup, diagnostics);
```

Comment:
>ReportMakeInvocationExpressionBadMemberKind(syntax, WellKnownMemberNames.CollectionInitializerAddMethodName, methodGroup, diagnostics); [](http://example.com/codeflow?start=20&length=135)

It is not obvious why this is necessary an error. For example, if a property returns a delegate, the property access is a valid primary expression for an invocation.